### DISCUSSION TITLE
IPv6 VIP and API listening

### DISCUSSION THREAD

**Discussion by imp1sh** [siderolabs/talos#9689] [General]
Hi,

I was wondering

a) Why is the API not binding to IPv6? I cannot call the talos API on port 50000 via IPv6, even when the host has a routable IPv6 address.
b) I want a vip set for the control plane nodes. This is already working but I cannot set an additional vip. I want an additional, because I also want an IPv6 VIP. This doesn't seem to be possible, or is it?

**Comment by smira** [comment]
API binds to all addresses, including IPv6, and works without issues. 

Talos VIP is used for Kubernetes API access only, not sure if there's any benefit in having both IPv4 and IPv6, but it's not supported at the moment.

**Comment by imp1sh** [comment]
When running the network in dual stack mode - which is what I'm doing until I can finally switch off IPv4 - it's the default behaviour to have every service reachable via both IPv4 and IPv6.
For now I will only run the IPv6 vip.
